src/tir/analysis/buffer_access_lca_detector.cc

```diff
@@ -150,6 +176,17 @@ class LCADetector : public StmtExprVisitor {
     }
   }
 
+  void UpdateWithBlockidx() {
+    for (const auto& it : buffer_lca_) {
+      if (GetRef<Buffer>(it.first).scope() == "global") {
```

Comment:
Use StorageScope here https://github.com/apache/tvm/blob/main/src/runtime/thread_storage_scope.h#L86

Consider move up to the ThreadScope as long as their rank is higher equal than threadScope.rank